pandas/core/frame.py

```diff
@@ -627,6 +627,12 @@ def __init__(
 
         manager = get_option("mode.data_manager")
 
+        # GH47215
+        if index is not None and isinstance(index, set):
```

Comment:
can we do these in `ensure_index` (or the Index constructor instead)

Reply:
I tried these, but it causes a bunch of tests to fail, because `.loc` calls `ensure_index` and we allow sets (but deprecated) there.